### PR TITLE
[Merged by Bors] - feat: port Algebra.GroupRingAction.Invariant

### PR DESCRIPTION
**Pull request by int-y1**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/GroupRingAction/Invariant.lean [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+variable {R}
+
+/-- A typeclass for subrings invariant under a `MulSemiringAction`. -/
+class IsInvariantSubring : Prop where
+  smul_mem : ∀ (m : M) {x : R}, x ∈ S → m • x ∈ S
+#align is_invariant_subring IsInvariantSubring
+
+instance IsInvariantSubring.toMulSemiringAction [IsInvariantSubring M S] : MulSemiringAction M S
+    where
```
Comment (dupuisf) [inline Mathlib/Algebra/GroupRingAction/Invariant.lean:33]:
```suggestion
instance IsInvariantSubring.toMulSemiringAction [IsInvariantSubring M S] : MulSemiringAction M S where
```

File: Mathlib/Algebra/GroupRingAction/Invariant.lean [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+variable (U : Subring R') [IsInvariantSubring M U]
+
+/-- The canonical inclusion from an invariant subring. -/
+def IsInvariantSubring.subtypeHom : U →+*[M] R' :=
+  { U.subtype with map_smul' := fun _ _ ↦ rfl }
+#align is_invariant_subring.subtype_hom IsInvariantSubring.subtypeHom
+
+-- Porting note: changed `coe` to `Subtype.val`
+@[simp]
+theorem IsInvariantSubring.coeSubtype_hom :
```
Comment (dupuisf) [inline Mathlib/Algebra/GroupRingAction/Invariant.lean:60]:
```suggestion
theorem IsInvariantSubring.coe_subtypeHom :
```

File: Mathlib/Algebra/GroupRingAction/Invariant.lean [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    (IsInvariantSubring.subtypeHom M U : U → R') = Subtype.val := rfl
+#align is_invariant_subring.coe_subtype_hom IsInvariantSubring.coeSubtype_hom
+
+-- Porting note: added `toRingHom`
+@[simp]
+theorem IsInvariantSubring.coe_subtype_hom' :
```
Comment (dupuisf) [inline Mathlib/Algebra/GroupRingAction/Invariant.lean:66]:
```suggestion
theorem IsInvariantSubring.coe_subtypeHom' :
```

**Review by dupuisf — COMMENTED** [review]
Thanks!

bors d+

**Comment by ChrisHughes24** [comment]
bors merge

**Pull request closed without merging**